src/update.rs

```diff
@@ -259,7 +283,7 @@ macro_rules! maybe_update_map {
 
 unsafe impl<K, V, S> Update for HashMap<K, V, S>
 where
-    K: Update + Eq + Hash,
+    K: Eq + Hash,
```

Comment:
I'd expect this bound to be necessary as otherwise we could store non update keys in the map? That is `HashMap<&'db (), ()>`

Reply:
Oh right. It's sort of annoying because we can't use the "Fallback" trick for containers. Or I just don't see how

Reply:
The fallback should apply right? I guess the issue is that we won't do a fine grained update on the elements within but instead on the thing as a whole?

Reply:
Right, it does apply if both key and value are static. It doesn't apply if the `value` requires `Update` but the `Key` otherwise wouldn't

Reply:
The bounds don't apply the fallback trick, right...